src/pykeen/pipeline.py

```diff
@@ -663,6 +674,17 @@ def pipeline(  # noqa: C901
         validation_triples_factory=validation_triples_factory,
     )
 
+    # evaluation restriction to a subset of entities/relations
```

Comment:
if  the restriction list is none, it should just pass through (just in case there's an issue with `TriplesFactory.new_with_restriction()`)

Reply:
Something like this might work:

```python
    testing_triples_factory, validation_triples_factory = [
        None if factory is None else
        factory.new_with_restriction(
            entities=restrict_evaluation_to_entities,
            relations=restrict_evaluation_to_relations,
        ) if restrict_evaluation_to_entities or restrict_evaluation_to_relations else
        factory
        for factory in (testing_triples_factory, validation_triples_factory)
    ]
```

Reply:
5e08e34